editor/src/clj/editor/editor_extensions.clj

```diff
@@ -368,6 +372,58 @@
           (catch Throwable e
             (throw (LuaError. ^String (ex-message e)))))))))
 
+(def ext-open-external-file-fn
+  (rt/suspendable-lua-fn open-external-file [{:keys [rt]} lua-string]
+    (let [file-name (rt/->clj rt coerce/string lua-string)]
+      (future/io
+        (try
+          (.open ui/desktop (io/file file-name))
+          (catch Throwable e
+            (throw (LuaError. ^String (ex-message e)))))))))
+
+;; region json
+
+(def json-decode-options-coercer
+  (coerce/hash-map :opt {:all coerce/boolean}))
+
+(def ext-json-decode
+  (letfn [(decode [^LuaString lua-string opts]
+            (with-open [r (-> lua-string .toInputStream io/reader (PushbackReader. 64))]
+              (if (:all opts)
+                (let [eof-value r]
+                  (loop [acc (transient [])]
+                    (let [ret (json/read r :eof-error? false :eof-value eof-value)]
+                      (if (identical? ret eof-value)
+                        (persistent! acc)
+                        (recur (conj! acc ret))))))
+                (json/read r))))]
+    (rt/lua-fn ext-json-decode
+      ([_ lua-string]
+       (decode lua-string nil))
+      ([{:keys [rt]} lua-string lua-options]
+       (decode lua-string (rt/->clj rt json-decode-options-coercer lua-options))))))
+
+(def json-value-coercer
+  (coerce/recursive
+    #(coerce/one-of
+       coerce/null
+       coerce/boolean
+       coerce/number
+       coerce/string
+       (coerce/vector-of % :min-count 1)
+       (coerce/map-of coerce/string %))))
+
+(def ext-json-encode
+  (rt/lua-fn ext-json-decode
+    ([{:keys [rt]} lua-value]
+     (json/write-str (rt/->clj rt json-value-coercer lua-value)))))
```

Comment:
For `ios-deploy`, we need to `decode` json. I added `encode` for completeness, even though the editor scripts do not use it.